src/patterns/ec2-app/base.ts

```diff
@@ -377,6 +449,115 @@ export class GuEc2App extends Construct {
+      // https://github.com/guardian/cognito-gatekeeper.
+      const authLambda = new GuLambdaFunction(scope, "auth-lambda", {
+        app: app,
+        memorySize: 128,
+        handler: "devx-cognito-lambda-amd64-v1",
+        runtime: Runtime.GO_1_X,
+        fileName: "devx-cognito-lambda-amd64-v1.zip",
+        bucketNamePath: NAMED_SSM_PARAMETER_PATHS.OrganisationDistributionBucket.path,
+        architecture: Architecture.X86_64,
+        environment: {
+          ALLOWED_GROUPS: allowedGroups.join(","),
+          GATEKEEPER_FUNCTION_ARN: gatekeeperFunctionArn,
+        },
+      });
+
+      authLambda.addToRolePolicy(
+        new PolicyStatement({
+          effect: Effect.ALLOW,
+          actions: ["lambda:InvokeFunction"],
+          resources: [gatekeeperFunctionArn],
+        })
+      );
+
+      // Cognito user pool. We require both lambdas: pre-sign-up runs the first
+      // time a user attempts to authenticate (before they exist in the User
+      // Pool); pre-auth runs in subsequent authentication flows.
+      const userPool = new UserPool(this, "user-pool", {
+        lambdaTriggers: {
+          preAuthentication: authLambda,
+          preSignUp: authLambda,
+        },
+      });
+
+      const userPoolDomain = userPool.addDomain("domain", {
+        cognitoDomain: {
+          // Must be unique (as global across AWS)
+          domainPrefix: `com-gu-${app.toLowerCase()}-${scope.stage.toLowerCase()}`,
+        },
+      });
+
+      const clientIDSSMParameter = StringParameter.fromStringParameterName(
+        scope,
+        "client-id-parameter",
+        clientIDSSMPath
+      );
+
+      const userPoolIdp = new UserPoolIdentityProviderGoogle(scope, "google-idp", {
+        userPool: userPool,
+        clientId: clientIDSSMParameter.stringValue,
+        clientSecret: SecretValue.secretsManager(clientSecretSecretsManagerPath).toString(),
+        attributeMapping: {
+          email: ProviderAttribute.GOOGLE_EMAIL,
+          givenName: ProviderAttribute.GOOGLE_GIVEN_NAME,
+          familyName: ProviderAttribute.GOOGLE_FAMILY_NAME,
+          profilePicture: ProviderAttribute.GOOGLE_PICTURE,
+          custom: {
+            name: ProviderAttribute.GOOGLE_NAME,
+          },
+        },
+        scopes: ["openid", "email", "profile"],
+      });
+
+      const userPoolClient = userPool.addClient("alb-client", {
+        supportedIdentityProviders: [UserPoolClientIdentityProvider.GOOGLE],
+        generateSecret: true,
+        oAuth: {
+          callbackUrls: [`https://${props.googleAuth.domain}/oauth2/idpresponse`],
+        },
+
+        // Note: id and access validity token validity cannot be less than one
+        // hour (this is the cognito cookie duration). To quickly invalidate
+        // credentials, disable the user in Cognito.
+        idTokenValidity: Duration.hours(1),
+        accessTokenValidity: Duration.hours(1),
+        refreshTokenValidity: Duration.days(7),
```

Comment:
TODO - we should probably support overriding these as some projects may want to be more aggressive on durations. Though note, allowing customisation allows less secure configurations here - there is an argument to just use the same settings everywhere.

Reply:
Worth moving this TODO into the file?

Reply:
(will do as a follow-up depending on requests from teams)